src/test/compile-fail/E0369.rs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+    let string = "Hello " + "World" + "!";
+    //~^ ERROR 12:18: 12:26: binary operation `+` cannot be applied to type `&'static str` [E0369]
+    //~| NOTE You can't use `+` between a `&str` and `&str`
+    //~| HELP Your first argument should be a `String`, not `&str`. Try using to_string()
+    //~| SUGGESTION let string = "Hello ".to_string() + "World" + "!"
+    //~| NOTE an implementation of `std::ops::Add` might be missing for `&'static str`
```

Comment:
@nrc - I think he's going to move to a UI test, so we won't need them (and we'll be able to see what the actual error will look like)